### ISSUE TITLE
JSON-RPC call vhost_create_nvme_controller doesn't seem to work for vhost-nvme

### ISSUE THREAD

**Issue by junehyuk-boo** [spdk/spdk#1191]
I have been trying to use the SPDK vhost-user-nvme, which is explained at the spdk.io documentation: [https://spdk.io/doc/vhost.html](https://spdk.io/doc/vhost.html)

However, when I was trying to set up the vhost-nvme's controller, I does not seem to work.

I was setting the environment to my server by the following:

```
# HUGEMEM=4096 scripts/setup.sh
# app/vhost/vhost -S /var/tmp -m 0x1 &
# ./scripts/rpc.py vhost_create_nvme_controller --cpumask 0x1 vhost.0 16
```

I also tried to set up the bdev controller before creating the vhost-nvme controller (like the vhost-scsi and the vhost-blk) by
`# sudo ./scripts/rpc.py bdev_nvme_attach_controller -b NVMe0 -t PCIe -a 0000:0a:00.0`
and then tried to create the vhost-nvme controller, but it does not work either.

What do I have to do to test the vhost-user-nvme?

## Expected Behavior

According to the document [https://spdk.io/doc/jsonrpc.html](https://spdk.io/doc/jsonrpc.html), I was expecting the rpc call just returns with creating an empty vhost-nvme controller

## Current Behavior

The vhost_create_nvme_controller returns as follows:
```
request:
{
  "ctrlr": "vhost.0",
  "io_queues": 8,
  "cpumask": "0x1",
  "method": "vhost_create_nvme_controller",
  "req_id": 1
}
Got JSON-RPC error response
response:
{
  "code": -32601,
  "message": "Method not found"
}
```
I have checked that in the rpc.py file, the request itself is implemented, for example, to have certain arguments.

I am guessing that the SPDK's rpc client does not support the method vhost_create_nvme_controller.

## Steps to Reproduce

1. HUGEMEM=4096 scripts/setup.sh
2. app/vhost/vhost -S /var/tmp -m 0x1 &
3. ./scripts/rpc.py vhost_create_nvme_controller --cpumask 0x1 vhost.0 16

## Context (Environment including OS version, SPDK version, etc.)

I am using the SPDK v20.04 with commit sha: f17e02a01, which was the latest SPDK master for me
I am also using ubuntu 18.04.3 with kernel version 5.3.0

**Comment by changpe1** [comment]
@junehyuk-boo Thanks for submitting, vhost-nvme is only compiled "with-internal-vhost-lib", we will not include vhost-nvme by default, because it needs a extra QEMU patch which doesn't in QEMU mainline branch.

**Comment by junehyuk-boo** [comment]
Thank you for the kind response.
I have configured the spdk with the "with-internal-vhost-lib" option, and for me the VM with vhost-user-nvme seems to work fine with the patched qemu (version 3.0.0).
It was very helpful, and if there are no other issues, I think I can close this.
